pkg/endpoint/restore.go

```diff
@@ -76,7 +76,7 @@ func ReadEPsFromDirNames(ctx context.Context, owner regeneration.Owner, policyGe
 				logfields.EndpointID: epDirName,
 			})
 			fullDirName := filepath.Join(basePath, epDirName)
-			scopedLog.Warning(fmt.Sprintf("Found incomplete restore directory %s. Removing it...", fullDirName))
+			scopedLog.Info(fmt.Sprintf("Found incomplete restore directory %s. Removing it...", fullDirName))
```

Comment:
The biggest behavioural shift in this PR is executing `Endpoint.aliveCancel()` early on in the `Endpoint.Stop()` process. Before, it seems like Cilium would happily go on to compile, rewrite and (try to) attach endpoint programs even though the CNI already gave up and the veths were long gone. This is more optimal, but could also surface a few new behaviours, and perhaps this incomplete dir is related. I assume dirs are left behind if endpoint compilation is cancelled, which sounds pretty normal to me. And after all, the agent cannot prevent itself from being terminated, so whatever on-disk state there is, it needs to tolerate.

I haven't put much thought into this one in particular, but isn't an incomplete dir just one with a `_next` or `_next_fail` suffix? The log message simply says "hey, I'm cleaning up". Also relatively normal. If the agent couldn't continue for whatever reason, the appropriate action would be a `Fatal()`, but I don't think that's the case here. Hence, info.

To me, the distinction between info and error lies between something needing operator intervention or not. From an operator POV, `level=error` in production should trigger an alert and cause someone to get paged. If not, what is its meaning?

Philosophical discussion aside, this case in particular sounds pretty expected, and there's little the agent can do. I think the pre-restart agent logs would contain the appropriate entries pertaining to this particular failure, unless it was SIGKILLed, but that sounds rather unlikely here.

Reply:
> I haven't put much thought into this one in particular, but isn't an incomplete dir just one with a `_next` or `_next_fail` suffix?

Yes. Hopefully there's still a regular directory without the suffix there at that time to allow a safe restore. So yeah I guess that's fine to just clean up. Agree that it should handle the case where the agent is just prematurely terminated; there's not much else we could do there.

> To me, the distinction between info and error lies between something needing operator intervention or not. From an operator POV, level=error in production should trigger an alert and cause someone to get paged. If not, what is its meaning?

:+1: but also the above is referring to `level=error` rather than `level=warning`. One thing I was thinking about is that if we can catch any sort of "unusual but recoverable" cases in CI by logging warnings then catching those in CI, then that can give us signal about areas to look closer, possibly even catching bugs before we release them.

But yeah I guess this is fine to just log at Info level, it's at least a signal in case there is anything weird going on here but there's not any strong signal that additional administrator intervention is required. :+1: